### ISSUE TITLE
[Feature] Make retina agent optional, configurable component

### ISSUE THREAD

**Issue by grzesuav** [Azure/AKS#4519]
**Is your feature request related to a problem? Please describe.**
I want to have ability to disable retina while keeping AKS managed protmethues, reasons are in https://github.com/Azure/AKS/issues/4508

**Describe the solution you'd like**
Option to toggle retina

**Describe alternatives you've considered**
We have a webhook supressing retina described in https://github.com/Azure/AKS/issues/4508#issuecomment-2310249153 but I want to have controll over that via AKS

**Additional context**
N/A

**Comment by grzesuav** [comment]
this is related to https://github.com/microsoft/retina/issues/655, which is not yet fixed. Even when it is fixed, I do not see any reason to install retina by default when the only thing I want are control-plane metrics. Unless AKS can provide other way to have control plane metrics

**Comment by vakalapa** [comment]
Fair enough, we will discuss internally and get back soon on this feature ask.

**Comment by lucalamalfa91** [comment]
Ehy @grzesuav we had the same issue, but in our case 

To prevent a DaemonSet from running (similar to scaling it down to zero replicas, which isn't natively supported for DaemonSets), you can configure it with a nodeSelector pointing to a non-existing label. This will effectively stop the DaemonSet from scheduling any pods.

Use the following command to update the DaemonSet with a non-existent nodeSelector:

`kubectl -n kube-system patch daemonset retina-agent -p '{"spec": {"template": {"spec": {"nodeSelector": {"non-existing": "true"}}}}}'`

This approach ensures that the DaemonSet is paused without disabled all the monitoring plugin as mentioned [here](https://github.com/Azure/AKS/issues/4508#issuecomment-2310731004)

**Comment by grzesuav** [comment]
Yes, I am aware, we have mutating webhook doing that as direct deamonset operations will be reverted, described this approach in the other issue.

**Comment by avi-sarvam** [comment]
@lucalamalfa91 We are also experiencing a significant increase in storage costs, specifically due to the networkobservability_tcp_connection_remote metric in the InsightsMetrics table. Despite applying the configuration from [ama-metrics-settings-configmap.yaml](https://github.com/Azure/prometheus-collector/blob/main/otelcollector/configmaps/ama-metrics-settings-configmap.yaml) and setting networkobservabilityRetina = false, we still observe this metric being scraped and ingested into the InsightsMetrics table. @vakalapa It would be great to have the requested feature implemented.

Additionally, @grzesuav, you mentioned that the nodeSelector operation will be reverted. Could you please share the steps to create a mutating webhook ? This would allow me to prevent the retina-agent pods from scheduling and in turn stop the scraping of this metric, which is causing exceptional costs, Thanks!

**Comment by grzesuav** [comment]
@avi-sarvam 
I implemented (actually extended one of the existing webhook we have) mutation webhook for DaemonSet with specific `app` label (in `kube-system`) namespace - https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/ .

I am using controller runtime for that - https://github.com/kubernetes-sigs/controller-runtime/blob/main/examples/builtins/mutatingwebhook.go however it might be more convenient to start with kubebuilder to generate boilerplate - https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation .

Then I just added selector team with some imaginary condition to met, which none of our node satisfy

**Comment by paulgmiller** [comment]
@avi-sarvam we hotfixed a metric cardinality issue that probably should have drastically reduced networkobservability_tcp_connection_remote (I believe checking with others) 

Curiosu to know if you're better or if you or others are stiil seeing Prometheus stress.  We're still considering disablement here but more ideal is to make this cheap enough no one cares it is running.

**Comment by neaggarwMS** [comment]
@grzesuav the high storage cost issue has been mitigated. We have fixed the networkobservability_tcp_connection_remote to not add remote ip/port info that can go unbounded. Apologies for the inconvenience for the same. Also please note these networking metrics that running as default with 1.29 are basic networking insights aggregated per node. They were added for full stack dashboard debugging/observability along with other useful control plane/cpu/memory metrics.

**Comment by MrImpossibru** [comment]
New issue found with it: Windows deployment resources requests are very high.
It's like nodes with 2 cores and 8 gigs of ram are useless - empty node even can't run all system containers.

Linux retina-agent yaml:
resources:
limits:
cpu: 500m
memory: 300Mi
requests:
cpu: 100m
memory: 200Mi

Windows retina-agent-win yaml:
resources:
limits:
cpu: 500m
memory: 300Mi
requests:
cpu: 500m
memory: 300Mi
